providers/imap/src/main/java/org/eclipse/angus/mail/iap/ResponseInputStream.java

```diff
@@ -122,7 +127,11 @@ public ByteArray readResponse(ByteArray ba) throws IOException {
                 int avail = buffer.length - idx; // available space in buffer
                 if (count + incrementSlop > avail) {
                     // need count-avail more bytes
-                    ba.grow(Math.max(minIncrement, count + incrementSlop - avail));
+                    int amountToGrow = Math.max(minIncrement, count + incrementSlop - avail);
+                    if (INPUT_STREAM_BUFFER_MAX_SIZE > 0 && buffer.length + amountToGrow > INPUT_STREAM_BUFFER_MAX_SIZE) {
```

Comment:
Do an unsigned compare.
```suggestion
                    if((buffer.length + amountToGrow)  - INPUT_STREAM_BUFFER_MAX_SIZE > 0) {
```

Or use Integer.compareUnsigned.